src/parser/jobs/brd/modules/PitchPerfect.js

```diff
@@ -464,7 +468,11 @@ export default class PitchPerfect extends Module {
 				stacksUsedInCurrentWM = 0
 				castsInCurrentWM = []
 			}
-			if (this.downtime.isDowntime(pp.lastTickOnEnemy + DOT_TICK_FREQUENCY + ANIMATION_LOCK)) {
+			const nextPPTick = pp.lastTickOnEnemy + DOT_TICK_FREQUENCY + ANIMATION_LOCK
+			if (
+				nextPPTick > this.parser.eventTimeOffset + this.parser.pull.duration
+				|| this.downtime.isDowntime(nextPPTick)
+			) {
```

Comment:
figures the end of the fight is about the same as downtime as far as forgiven early PPs goes